b2blaze/models/file_list.py

```diff
@@ -59,53 +90,140 @@ def _update_files_list(self, retrieve=False):
+        params = {
+            'bucketId': self.bucket.bucket_id,
+            'maxFileCount': 10000
+        }
+
+        # Limit files
+        if limit:
+            params['maxFileCount'] = limit
+
+        while new_files_to_retrieve:
 
-            response = self.connector.make_request(path, method='post', params=params)
+            response = self.connector.make_request(path=path, method='post', params=params)
             if response.status_code == 200:
-                file_json = response.json()
-                if len(file_json['files']) > 0:
-                    return B2File(connector=self.connector, parent_list=self, **file_json['files'][0])
+                files_json = response.json()
+                for file_json in files_json['files']:
+                    new_file = B2File(connector=self.connector, parent_list=self, **file_json)
+
+                    # Append file_id, file_name to lists
+                    file_name, file_id = file_json['fileName'], file_json['fileId']
+                    file_names.append(file_name)
+                    file_ids.append(file_id)
+                    
+                    # Add file to list keyed by file_id
+                    if file_id in file_versions:
+                        file_versions[file_id].append(new_file)
+                    else:
+                        file_versions[file_id] = [new_file]
+
+                if files_json['nextFileName'] is None:
+                    new_files_to_retrieve = False
                 else:
-                    raise B2FileNotFound('fileName - ' + file_name)
-            else:
-                raise B2RequestError(decode_error(response))
-        elif file_id is not None:
-            path = '/b2_get_file_info'
-            params = {
-                'fileId': file_id
-            }
-            response = self.connector.make_request(path, method='post', params=params)
-            if response.status_code == 200:
-                file_json = response.json()
-                return B2File(connector=self.connector, parent_list=self, **file_json)
+                    params['startFileName'] = files_json['nextFileName']
             else:
-                raise B2RequestError(decode_error(response))
+                raise B2Exception.parse(response)
+        return {'file_names': file_names, 'file_versions': file_versions, 'file_ids': file_ids}
+
+
+    def _get_by_name(self, file_name):
+        """ Internal method, return single file by file name """ 
+        path = API.list_all_files
+        params = {
+            'prefix': b2_url_encode(file_name),
+            'bucketId': self.bucket.bucket_id
+        }
+
+        response = self.connector.make_request(path, method='post', params=params)
+        file_json = response.json()
+
+        # Handle errors and empty files
+        if not response.status_code == 200:
+            raise B2Exception.parse(response)
+        if not len(file_json['files']) > 0:
```

Comment:
```python
if len(file_json['files']) == 0:

# or

if not len(file_json['files']):
```

Reply:
@42B Nice, thanks for the improvement! Method #2 is nice, very clean. Those if len(.... ) statements always seem a bit unaesthetic to me.